docs/source/traits_user_manual/defining.rst

```diff
@@ -405,6 +405,57 @@ the table.
 |                  | *allow_none* = False, *adapt* = 'yes', \*\*\ *metadata*])|
 +------------------+----------------------------------------------------------+
 
+.. index:: Instance trait
+
+.. _instance:
+
+Instance
+::::::::
+One of the most fundamental and useful predefined trait types is Instance.
+Instance trait values are an instance of a particular class or its subclasses,
+as specified by the **klass** argument. **klass** can be either an instance of
+a class or a class itself (note this applies to all python classes, not
+necessarily just ``HasTraits`` subclasses). Further, **klass** can also be an
```

Comment:
I think we should drop the sentence about interfaces. Historical background: `Instance(ISomeInterface)` checks used to work differently from `Instance(SomeRandomClass)` checks - the former genuinely would check that the type of the assigned value had all the right methods. That's most of what the stuff in the "interfaces" section is discussing. (And yes, that section needs to be updated and revised, but one thing at a time ...)

So it's true that `klass` can be an interface, but in that case, in modern Traits, the check is still an `isinstance(assigned_object, ISomeInterface)`, just like it is for regular classes.

It's tricky, because use of interfaces here _is_ an important use-case, but I don't want to give the impression that interfaces are somehow treated differently from other classes - we're just using a simple `isinstance` check in all cases.